src/HasBinaryUuid.php

```diff
@@ -73,17 +73,90 @@ public static function decodeUuid(string $binaryUuid): string
 
     public function toArray()
     {
-        if (! $this->exists) {
-            return parent::toArray();
+        $uuidAttributes = $this->getUuidAttributes();
+
+        $array = parent::toArray();
+
+        // if (! $this->exists) {
+        // return $array;
+        // }
+
+        if (is_array($uuidAttributes)) {
+            foreach ($uuidAttributes as $attributeKey) {
+                if (array_key_exists($attributeKey, $array)) {
+                    $uuidKey = $this->getRelatedBinaryKeyName($attributeKey);
+                    $array[$attributeKey] = $this->{$uuidKey};
+                }
+            }
+        }
+
+        return $array;
+    }
+
+    public function getRelatedBinaryKeyName($attrib)
+    {
+        $suffix = $this->getUuidTextAttributeSuffix();
+
+        return "{$attrib}{$suffix}";
+    }
+
+    public function getAttribute($key)
+    {
+        if (($uuidKey = $this->uuidTextAttribute($key)) && $this->{$uuidKey} !== null) {
```

Comment:
To make this more readable, you should avoid assignment in an `if` statement:

```php
$uuidKey = $this->uuidTextAttribute($key);

if ($uuidKey && $this->{$uuidKey} !== null)
```

Reply:
Yes true